### DISCUSSION TITLE
Value storage for Actions

### DISCUSSION THREAD

**Discussion by john-jorgensen-vectorsolutions** [zitadel/zitadel#7807] [Q&A] [answered]
I have a use case where I need to call a protected API from within an action whenever a user authenticates through an external IDP. Authenticating users won't necessarily have access to this API, so the action will have client credentials it can use to authenticate and get an access token to call the API. To reduce the number of times this action needs to get an access token for my API, I would like to store the access token somewhere so future executions of the action could check for an existing token and use it.

Is there some way to store values during the execution of an action, where future executions of that action could read those values?

**Comment by hifabienne** [comment, marked as answer]
hei @jjorgen-vs as far as i know there is currently no option to store some values and get them in the actions.
However, we are currently implementing our actions v2, which work a little different than the current.
You will run the custom code on one of you systems and only tell zitadel an endpoint to trigger at which point.
In that case you will have more options about how to implement your actions and how to use values.

@stebenz @peintnermax please correct me if I stated something wrong, or add additional information

**Reply by gabriel-hoff-vectorsolutions** [reply]
Does this mean actions v2 will only call external APIs? Or will it also provide a way to execute custom JavaScript code?

**Reply by stebenz** [reply]
Yes, actions v2 are mainly to call external APIs, the execution of custom JavaScript is not planned yet.
